sdcflows/interfaces/bspline.py

```diff
@@ -288,116 +288,110 @@ def _run_interface(self, runtime):
 
 
 class _ApplyCoeffsFieldInputSpec(BaseInterfaceInputSpec):
-    in_data = InputMultiObject(
-        File(exist=True, mandatory=True, desc="input EPI data to be corrected")
-    )
+    in_data = File(exist=True, mandatory=True, desc="input EPI data to be corrected")
     in_coeff = InputMultiObject(
         File(exists=True),
         mandatory=True,
-        desc="input coefficients, after alignment to the EPI data",
-    )
-    in_xfms = InputMultiObject(
-        File(exists=True), desc="list of head-motion correction matrices"
+        desc="input coefficients as calculated in the estimation stage",
     )
-    ro_time = InputMultiObject(
-        traits.Float(mandatory=True, desc="EPI readout time (s).")
+    fmap2data_xfm = File(
+        exists=True,
+        desc="the transform by which the fieldmap can be resampled on the target EPI's grid.",
     )
-    pe_dir = InputMultiObject(
-        traits.Enum(
-            "i",
-            "i-",
-            "j",
-            "j-",
-            "k",
-            "k-",
-            mandatory=True,
-            desc="the phase-encoding direction corresponding to in_data",
-        )
+    in_xfms = File(exists=True, desc="list of head-motion correction matrices")
+    ro_time = traits.Float(mandatory=True, desc="EPI readout time (s).")
+    pe_dir = traits.Enum(
+        "i",
+        "i-",
+        "j",
+        "j-",
+        "k",
+        "k-",
+        mandatory=True,
+        desc="the phase-encoding direction corresponding to in_data",
     )
     num_threads = traits.Int(nohash=True, desc="number of threads")
+    approx = traits.Bool(
+        True,
+        usedefault=True,
+        desc=(
+            "reconstruct the fieldmap on it's original grid and then interpolate on the "
+            "rotated grid, rather than reconstructing directly on the rotated grid."
+        ),
+    )
 
 
 class _ApplyCoeffsFieldOutputSpec(TraitedSpec):
     out_corrected = OutputMultiObject(File(exists=True))
     out_field = OutputMultiObject(File(exists=True))
-    out_warp = OutputMultiObject(File(exists=True))
 
 
 class ApplyCoeffsField(SimpleInterface):
-    """Convert a set of B-Spline coefficients to a full displacements map."""
+    """
+    Undistort a target, distorted image with a fieldmap, formalized by its B-Spline coefficients.
+
+    Preconditions:
+
+    * We have a "target EPI" - a BOLD or DWI dataset (or even MPRAGE, same principle),
+      without having gone through HMC or SDC.
+    * We have also the list of HMC matrices that *accounts for* head-motion, so after resampling
+      the dataset through this list of transforms *the head does not move anymore*.
+    * We have estimated the fieldmap's coefficients
+    * We have the "fieldmap-to-data" affine transform that aligns the target dataset (e.g., EPI)
+      and the fieldmap's "magnitude" (phasediff et al.) or "reference" (pepolar, syn).
```

Comment:
Are we eventually going to add optional transformations from data to final target? That will be the ideal end state where the final resampling to any space includes the fieldmap shift in source space.

Reply:
I think this would be ideal. For that, I believe the best approach would be adding to nitransforms a dense deformation field that works in VOXEL coordinates. There will be some plumbing for composite transforms so that the voxel field is cast to a regular dense field using the (I believe as I write) reference image.

Once we get this right, and I feel it is almost there (just some testing lacking), in my opinion we'll be ready to beef up nitransforms to do that (and perhaps surface too?)